Libraries/src/Amazon.Lambda.Annotations/APIGateway/HttpResults.cs

```diff
@@ -0,0 +1,437 @@
+
+        // See comment in class documentation on the rules for serializing. If any changes are made in this method be sure to update
+        // the comment above.
+        private void FormatBody(object body)
+        {
+        // See comment at the top about .NET Standard 2.0
+#if NETSTANDARD2_0
+            throw new NotImplementedException();
+#else
+
+            if (body == null)
+                return;
+
+            if (body is string str)
+            {
+                _body = str;
+            }
+            else if (body is Stream stream)
+            {
+                _isBase64Encoded = true;
+                var buffer = ArrayPool<byte>.Shared.Rent((int)stream.Length);
+                try
+                {
+                    var readLength = stream.Read(buffer, 0, buffer.Length);
+                    _body = Convert.ToBase64String(buffer, 0, readLength);
+                }
+                finally
+                {
+                    ArrayPool<byte>.Shared.Return(buffer);
+                }
+            }
+            else if (body is byte[] binaryData)
+            {
+                _isBase64Encoded = true;
+                _body = Convert.ToBase64String(binaryData, 0, binaryData.Length);
+            }
+            else if (body is IList<byte> listBinaryData)
+            {
+                _isBase64Encoded = true;
+                _body = Convert.ToBase64String(listBinaryData.ToArray(), 0, listBinaryData.Count);
+            }
+            else
+            {
+                _body = JsonSerializer.Serialize(body);
+            }
+#endif
+        }
+
+        /// <summary>
+        /// Serialize the IHttpResult into the expect format for the event source.
+        /// </summary>
+        /// <param name="options"></param>
+        /// <returns></returns>
+        public Stream Serialize(HttpResultSerializationOptions options)
+        {
+        // See comment at the top about .NET Standard 2.0
+#if NETSTANDARD2_0
+            throw new NotImplementedException();
+#else
+            var stream = new MemoryStream();
+            if (options.Format == HttpResultSerializationOptions.ProtocolFormat.RestApi ||
+                (options.Format == HttpResultSerializationOptions.ProtocolFormat.HttpApi && options.Version == HttpResultSerializationOptions.ProtocolVersion.V1))
+            {
+                var response = new APIGatewayV1Response
+                {
+                    StatusCode = (int)_statusCode,
+                    Body = _body,
+                    MultiValueHeaders = _headers,
+                    IsBase64Encoded = _isBase64Encoded
+                };
+
+                JsonSerializer.Serialize(stream, response);
+            }
+            else
+            {
+                var response = new APIGatewayV2Response
+                {
+                    StatusCode = (int)_statusCode,
+                    Body = _body,
+                    Headers = ConvertToV2MultiValueHeaders(_headers),
```

Comment:
I'm not sure exactly where in the code it's happening, but when I try:
```
return HttpResults.BadRequest("Cannot divide by zero")
                    .AddHeader("key", "value1")
                    .AddHeader("key", "value2");
```
it results in `key: value1, value2` as expected.

But this:
```
return HttpResults.BadRequest("Cannot divide by zero")
                    .AddHeader("key", "value1")
                    .AddHeader("key", "value2")
                    .AddHeader("KEY", "VALUE3");
```
results in `key: VALUE3`. I was expecting `key: value1, value2, VALUE3` since the header keys should be case-insensitive, but let me know if my understanding is incorrect. I think separate `key` and `KEY` headers might be okay since multiple values could be comma-separated or on multiple lines (would have to double-check spec), but wasn't expecting the first two values to be wiped out.

Reply:
Updated the `AddHeader` method to do a `ToLower` on the name and added your example as a test case.